File: Questoes/1450D-RatingCompression.cpp
#include<bits/stdc++.h>

using namespace std;

void solve(){

	int n;
	cin >> n;
	
	int array[n], cnt[n+1];
	set<int> s;
	int ans[n];
	memset(cnt, 0, sizeof(cnt));
	
	for(int i =0; i<n; i++){
		cin >> array[i];
		cnt[array[i]]++;
		s.insert(array[i]);
		ans[i] = 0;
	}
	
	bool flag = true;
	int l = 0;
	int r = n-1;
	
	for(int i =1; i<=n; i++){
		if(flag && cnt[i] > 0)
			ans[i-1] = 1;
			
		if(cnt[i] != 1)
			flag = false;
		else if(array[l] == i)
			l++;
		else if(array[r] == i)
			r--;
		else
			flag = false;
	}
	
	if((int)s.size() == n)
		ans[n-1] = 1;
	
	for(int i=n-1; i>=0; i--)
		cout << ans[i];
	cout << endl;
	
}

int main(){
	
	int t;
	cin >> t;
	
	while (t--) solve();

	return 0;
}

File: Questoes/1450D-RatingCompression.py
t = int(input())

for tt in range(t):
	n = int(input())
	array = [int(x) for x in input().split()]
	cnt = [0] * (n+1)
	
	for e in array:
		cnt[e] += 1
	
	ans = []	
	flag = True
	l = 0
	r = n-1
	
	for i in range(1, n):
		if flag and cnt[i]:
			ans.append(1)
		else:
			ans.append(0)
		
		if cnt[i] != 1:
			flag = False
		elif array[l] == i:
			l += 1
		elif array[r] == i:
			r -= 1
		else:
			flag = False
	
	aux = set(array)
	if len(aux) == n:
		ans.append(1)
	else:
		ans.append(0)
	
	ans.reverse()
	for e in ans:
		print(e, end="")
	print()
